libs/sql-parser/src/main/java/io/crate/sql/tree/ColumnDefinition.java

```diff
@@ -31,52 +31,67 @@ public class ColumnDefinition<T> extends TableElement<T> {
 
     private final String ident;
 
-    @Nullable
-    private final T defaultExpression;
-
-    @Nullable
-    private final T generatedExpression;
-
-    private final boolean generated;
-
     @Nullable
     private final ColumnType<T> type;
 
     private final List<ColumnConstraint<T>> constraints;
 
     public ColumnDefinition(String ident,
-                            @Nullable T defaultExpression,
-                            @Nullable T generatedExpression,
                             @Nullable ColumnType<T> type,
                             List<ColumnConstraint<T>> constraints) {
-        this(ident, defaultExpression, generatedExpression, type, constraints, true, generatedExpression != null);
-    }
-
-    public ColumnDefinition(String ident,
-                            @Nullable T defaultExpression,
-                            @Nullable T generatedExpression,
-                            @Nullable ColumnType<T> type,
-                            List<ColumnConstraint<T>> constraints,
-                            boolean validate,
-                            boolean generated) {
         this.ident = ident;
-        this.defaultExpression = defaultExpression;
-        this.generatedExpression = generatedExpression;
-        this.generated = generated;
         this.type = type;
         this.constraints = constraints;
-        if (validate) {
-            validateColumnDefinition();
-        }
+        validateColumnConstraints(ident, type, constraints);
     }
 
-    private void validateColumnDefinition() {
-        if (type == null && generatedExpression == null) {
+    static <T> void validateColumnConstraints(String ident,
```

Comment:
I'm considering since this throws an: `io.crate.exceptions.SQLParseException: Column [col1]: multiple default constraints found`, if we should validate in `TableElementsAnalyzer` instead as the parser allows it but it's more an analysis validation.

On second thought it's a gray area, can also be an parser exception, what do you think?

Reply:
It's fine to leave it here. I agree with your argument, it's like restricting a leniency of the parser, thx!